File: AlgoritmosOrdenamiento/Comparaciones/src/BubbleSort.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <ctime>
#include <string>

using namespace std;

unsigned t0, t1;

void bubbleSort( int *arr, int size ) {
    for ( int i = size - 1; i > 0; i-- ) {    
        for ( int j = 0; j < i; j++ ) {
            if( arr[ j ] > arr[ j + 1 ]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    } 
}

int main() {

    char caracter;
    std::string path, name = "aleatorio";
    std::string pathS, out = "salida";

    int tam[] = {10000, 30000, 50000, 70000, 100000};

    for( int i = 0; i < 10; ++i ) {
        path = "Entradas/" + name + std::to_string(i + 1) + ".txt";
        pathS = "Salidas/BubbleSort/" + out + std::to_string(i + 1) + "cc.txt";
        cout << pathS << endl;
        ofstream fs( pathS );

        char* in = const_cast<char*>( path.c_str());
        
        for(int j = 0; j < 5; ++j ) {
            int *A = new int[tam[j]];
            ifstream file;
            string num;
            int num_int = 0;
            file.open( in, ios::in);
            int c = 0;
            while(!file.eof() && c <=tam[j]){
                getline(file, num);
                stringstream convert(num);
                convert >> num_int;
                A[c] = num_int;
                c++;
            }

            file.close();
            
            t0=clock();
            bubbleSort(A,tam[j]);
            t1 = clock();

            double time = (double( t1 - t0 ) / CLOCKS_PER_SEC);
            fs << tam[j] << " " << time <<endl;
            
            delete[] A;
        }
        fs.close();
    }
}

File: AlgoritmosOrdenamiento/Comparaciones/src/BubbleSort.py
from io import open
from time import time

def bubbleSort(arr):
   n = len(arr)
   for i in range(n):
      for j in range(0, n-i-1):
         if int(arr[j]) > int(arr[j+1]) :
            arr[j], arr[j+1] = arr[j+1], arr[j]

tam = [100000, 300000, 500000, 700000, 1000000]
times = [0,0,0,0,0]
f=open("Salidas/BubbleSort/salida1py.txt","w")

for j in range(5):            
    file = open("Entradas/aleatorio1.txt","r")
    A = file.readlines()
    file.close()

    B=[0]*tam[j]

    for i in range(tam[j]):
        B[i]=int(A[i])

    t0=time()
    bubbleSort(A) 
    tiempo = time()-t0
    tiempo = round(tiempo,3)

    A.clear()
    f.write(str(tam[j])+" "+str(tiempo)+"\n")